### ISSUE TITLE
jsx-jcs-no-undef triggers for globals like JSON or window

### ISSUE THREAD

**Issue by texttechne** [vkbansal/eslint-plugin-jsx-control-statements#14]
While `no-undef` doesn't warn about `JSON` being undefined, `jsx-jcs-no-undef` does warn about that.

Please note, that I don't have any environment setting specifically for jsx-control-statements enabled. Might be related to #11?

**Comment by AlexGilleran** [comment]
"Environment" in ESLint basically just means a bunch of globals... if you're using the recommended config for eslint-plugin-jsx-control-statements and/or you're not seeing an error for every control statement you should already have it.

Have you set up "browser" as an env in eslintrc or equivalent? Should look something like this:

```
    "env": {
        "browser": true
    }
```

**Comment by texttechne** [comment]
yup, got it. So you don't get these errors for any kind of built-in stuff, like `console`, `window`, `JSON`, ...?

**Comment by AlexGilleran** [comment]
No :/. Although if I comment out the `"browser": true` line I do...

**Comment by AlexGilleran** [comment]
Looking at your other issue, you've got this:

```
    # "no-undef": 0,
    # "jsx-control-statements/jsx-jcs-no-undef": 1,
```

Is it possible that you normally have those set so no-undef just has no errors in general?

**Comment by AlexGilleran** [comment]
Sorry about all the "maybe it's your config?" answers, they're a cop-out but unfortunately I've got no other theories 😕

**Comment by stoikerty** [comment]
I think the error I get is related to this issue. With the following setting

``` json
  "globals": {
    "app": true
  },
```

I get an error when I use `app` somewhere.

```
ERROR in ./src/client/app.jsx
Module build failed: TypeError: (options.globals || []).reduce is not a function
    at new Config (/my_project/node_modules/eslint/lib/config.js:183:44)
    at CLIEngine.executeOnText (/my_project/node_modules/eslint/lib/cli-engine.js:750:28)
    at lint (/my_project/node_modules/eslint-loader/index.js:42:18)
    at Object.module.exports (/my_project/node_modules/eslint-loader/index.js:166:3)
```

I thought it was related to it not being an array (even though the previous declaration is [the correct way to define them](http://eslint.org/docs/user-guide/configuring#specifying-globals)), nonetheless I changed it to

```
  "globals": [
    "app"
  ],
```

But now I get this warning, so it must be ignoring the globals.

```
warning  'app' is not defined  jsx-control-statements/jsx-jcs-no-undef
```

Maybe this helps?
